Uchu.World/Objects/Components/Server/PhysicsComponent.cs

```diff
@@ -53,8 +55,58 @@ public void SetPhysics(PhysicsObject physics)
             physics.Associate = this;
 
             physics.OnCollision = HandleCollision;
-            
+
             Physics = physics;
+
+            // This is for POIs with a POI key in the settings, used for example everywhere in AG except for
+            // Monument Orange Path (AG_Mon_3).
+            // Not all POIs are handled this way, for example those in GF are done using triggers.
+            if (GameObject.Settings.TryGetValue("POI", out var poiGroup))
+            {
+                Logger.Information($"Registered POI location {poiGroup}");
+
+                Listen(GameObject.GetComponent<PhysicsComponent>().OnEnter, async component =>
+                {
+                    if (!(component.GameObject is Player player)) return;
+
+                    Logger.Debug($"{player.Name} entered {(string) poiGroup}");
+
+                    var inventory = player.GetComponent<MissionInventoryComponent>();
+                    await inventory.DiscoverAsync((string) poiGroup);
+                });
+            }
+        }
+
+        public void SetPhysicsByPath(string path) // We can't read HKX so this is basically just a bodge
```

Comment:
How complicated is HKX parsing? Should we make a GitHub Issue to look into this later?

Reply:
https://github.com/SimonNitzsche/OpCrux-Server/blob/master/src/FileTypes/HKXFile/hkxFile.cpp
Not worth the effort for the few simple shapes that are relevant, imo.

Reply:
That is a massive mess by Simon, Ignore that. It isn't that complicated once you work out the structs you need.

Reply:
Parsing HKX is too much effort, LU only uses basic shapes.

Reply:
It isn't, the structs for the objects are public information and parsing the header isn't that bad. You can even auto generate readers for all of the object structs.

Reply:
> It isn't, the structs for the objects are public information and parsing the header isn't that bad. You can even auto generate readers for all of the object structs.

Just throwing out there that it possibly not being too bad (don't know if it is or not, never worked with it), doesn't actually respond to Wincent's message: The question is not if it's doable, but whether it's actually easier than the alternative.
Don't have enough information one way or another, but just wanted to note a false dichotomy.